### PR TITLE
Update libcalico-go and fix tests

### PR DESCRIPTION
**Pull request by caseydavenport**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: pkg/controllers/networkpolicy/policy_controller.go [hunk 1]
```diff
@@ -238,8 +238,8 @@ func (c *policyController) syncToDatastore(key string) error {
 		}
 
 		clog.Infof("Update NetworkPolicy in Calico datastore with resource version %s", p.ResourceVersion)
-		p.ResourceVersion = gp.ResourceVersion
-		_, err = c.calicoClient.NetworkPolicies().Update(c.ctx, &p, options.SetOptions{})
+		gp.Spec = p.Spec
```
Comment (robbrockbank) [inline pkg/controllers/networkpolicy/policy_controller.go]: Do we care about not copying the annotations or labels?

File: pkg/controllers/namespace/namespace_controller.go [hunk 2]
```diff
@@ -241,10 +241,10 @@ func (c *namespaceController) syncToDatastore(key string) error {
 			return nil
 		}
 
-		// The profile already exists - update it.
-		p.ResourceVersion = gp.ResourceVersion
+		// The profile already exists - update it to have the new spec.
```
Comment (robbrockbank) [inline pkg/controllers/namespace/namespace_controller.go]: nit:  should `gp` be `prof` or `pr` or something like that?

File: pkg/controllers/networkpolicy/policy_controller.go (hunk 1 above)
Comment (caseydavenport) [reply pkg/controllers/networkpolicy/policy_controller.go]: yeah good point, we probably do want to do that.

File: pkg/controllers/namespace/namespace_controller.go (hunk 2 above)
Comment (caseydavenport) [reply pkg/controllers/namespace/namespace_controller.go]: gp I think stands for `getprofile`, as in the profile we got from the datastore.

File: pkg/controllers/networkpolicy/policy_controller.go (hunk 1 above)
Comment (caseydavenport) [reply pkg/controllers/networkpolicy/policy_controller.go]: On second thought, I'm not so sure.


It's also a larger change than it seems and requires a new libcalico, so I'm not going to do it in this PR.

**Pull request merged**